certora/specs/CER-44-VaultStatusCheck/CER-78-VaultStatusCheck-scheduling.spec

```diff
@@ -0,0 +1,40 @@
+import "../utils/IsMustRevertFunction.spec";
+
+// CER-78: Vault Status Check scheduling
+// Only the Vault is allowed require the check for itself
+methods {
+    function EthereumVaultConnector.requireVaultStatusCheckInternal(address vault) internal with (env e) => requireVaultStatusCheckOnlyCalledBySelf(e, vault);
+    // For the calls that allow deferred checks (call, batch, 
+    // controlCollateral), we cannot prove that the
+    // when the deferred checks are executed at the end of the deferred check
+    // call that the set of vault addresses did not already contain
+    // some vault address other than e.msg.sender.
+    // These summaries exclude the deferred checks from the rule.
```

Comment:
couldn't we have a hook on `vaultStatusChecks.insert` and check that we're only inserting `msg.sender`?

Reply:
Hi Kasper. Sure I think that is an improvement on this rule, so I implemented your idea. This has to be implemented as hooks on the lower-level values of vaultStatusChecks (the actual set elements and firstElement) because: 1) hooks can attach to variables but not functions (though function summaries can do something similar for functions) 2) we can't use a summary on insert because we need to distinguish between insert on vaultStatusChecks and other `SetStorage`s. As a result using this hook we actually prove something slightly stronger, that `vaultStatusCheck` is not _modified_ (beyond just inserted into) for any address other than `e.msg.sender`.

There is however a limitation related to `forgiveVaultStatusCheck` for the time being and a comment about that. See what you think of this.